mcfly.zsh

```diff
@@ -0,0 +1,103 @@
+#!/bin/zsh
+
+# Ensure stdin is a tty
+[[ -t 0 ]] || return
+
+# Avoid loading this file more than once
+if [[ "$__MCFLY_LOADED" == "loaded" ]]; then
+  return 0
+fi
+__MCFLY_LOADED="loaded"
+
+# Ensure HISTFILE exists.
+export HISTFILE="${HISTFILE:-$HOME/.zsh_history}"
+if [[ ! -r "${HISTFILE}" ]]; then
+  echo "McFly: ${HISTFILE} does not exist or is not readable. Please fix this or set HISTFILE to something else before using McFly."
+  return 1
+fi
+
+# MCFLY_SESSION_ID is used by McFly internally to keep track of the commands from a particular terminal session.
+export MCFLY_SESSION_ID=$(dd if=/dev/urandom bs=256 count=1 2> /dev/null | env LC_ALL=C tr -dc 'a-zA-Z0-9' | head -c 24)
+
+setopt extended_history       # record timestamp of command in HISTFILE
+setopt hist_expire_dups_first # delete duplicates first when HISTFILE size exceeds HISTSIZE
+setopt hist_ignore_dups       # ignore duplicated commands history list
+setopt hist_ignore_space      # ignore commands that start with space
+setopt inc_append_history     # add commands to HISTFILE in order of execution
+setopt share_history          # share command history data
+setopt interactive_comments   # allow comments in interactive shells (like Bash does)
```

Comment:
Are all these options required for using mcfly?
If they are, I'd suggest clearly documenting that to avoid surprising users when you change their options.
If they're not, better leave the options the users set themselves.

Reply:
I've now done some testing— I think the only one I actually need is `interactive_comments`.